src/microbe_stage/AgentProjectile.cs

```diff
@@ -47,7 +51,12 @@ public override void _Process(float delta)
 
         FadeTimeRemaining -= delta;
         if (FadeTimeRemaining <= 0)
-            Destroy();
+            OnDestroyed();
+    }
+
+    public void OnDestroyed()
+    {
+        this.DetachAndQueueFree();
```

Comment:
Shouldn't this be setting AliveMarker to false instead? I think this could possibly conflict with `DestroyDetachAndQueueFree`.

Reply:
It's very wrong to call DetachAndQueueFree in a OnDestroyed method.